common/auto-updater/http-downloader.cpp

```diff
@@ -0,0 +1,186 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+
+
+#ifdef ENABLE_RS_AUTO_UPDATER
+#include <sstream>
+#include <iostream>
+#include <thread>
+#include <atomic>
+#include <mutex>
+#include <curl/curl.h>
+#include <curl/easy.h>
+#include "types.h"
+#endif // ENABLE_RS_AUTO_UPDATER
+
+#include "http-downloader.h"
+
+
+namespace rs2
+{
+#ifndef ENABLE_RS_AUTO_UPDATER
+    // Dummy functions
+    http_downloader::http_downloader() {};
```

Comment:
explixic ctor/dtor can be removed

Reply:
I thought it is best to specifically place them for readability.  
When someone looks at this code he will know that the Ctor and Dtor are implemented empty in purpose with the flag off.

Do you still feel I should drop them?

Reply:
Up to you

Reply:
Then It remains.